comp/core/flare/component.go

```diff
@@ -0,0 +1,39 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+// Package flare implements a component to generate flares from the agent.
+//
+// A flare is a archive containing all the information necessary to troubleshoot the Agent. When openeing a support
+// ticket a flare might be requested. Flares contains the Agent logs, configurations and much more.
+package flare
+
+import (
+	pkgFlare "github.com/DataDog/datadog-agent/pkg/flare"
+	"github.com/DataDog/datadog-agent/pkg/util/fxutil"
+	"go.uber.org/fx"
+)
+
+// team: agent-shared-components
+
+// Component is the component type.
+type Component interface {
+	// Create creates a new flare locally and returns the path to the flare file.
+	Create(local bool, distPath, pyChecksPath string, logFilePaths []string, pdata pkgFlare.ProfileData, ipcError error) (string, error)
```

Comment:
This is a 2 item issue:
- First this keep the API between the component and the package consistent while we migrate to components. I also tried to not change the flare logic to much.
- This can actually be changed through the CLI when creating a flare locally (using env variables).
- It's ease testing between the flare comp and the flare package.

We should definitely change the API once we're done moving to component. Then we could clean the test, remove the flare package, clean the API ...